src/interfaces/messageData.ts

```diff
@@ -24,6 +25,39 @@ export interface ICognigyData {
 	controlCommands?: ISetRatingControlCommand
 }
 
+export interface IPluginDatepicker {
+	type: "date-picker";
+	data: {
+		eventName?: string;
+		locale?: string;
+		enableTime?: string | boolean;
+		defaultDate?: string,
+		mode?: "range" | "single" | "multiple";
+		wantDisable?: string | boolean;
+		enable_disable?: string[] | boolean | null;
+		function_enable_disable?: string;
+		minDate?: string;
+		maxDate?: string;
+		openPickerButtonText?: string;
+		cancelButtonText?: string;
+		submitButtonText?: string;
+		time_24hr?: string | boolean;
+		dateFormat?: string;
+		defaultHour?: number;
+		defaultMinute?: number;
+		enableSeconds?: number;
+		hourIncrement?: number;
+		minuteIncrement?: number;
+		noCalendar?: boolean;
+		weekNumbers?: boolean;
+	};
+}
+
+export interface IPluginDefault {
+	type: "default";
+	data: object;
+}
```

Comment:
Would it be possible to use something like that?

```typescript
export interface IPluginDefault<T = PluginData> {
    type: "default";
    data: T;
}
```

Reply:
I will try. but I will first try to remove it, it was temporary. We should know which exact plugins we have. default option should not exists.